azurerm/resource_arm_application_insights_analytics_item_test.go

```diff
@@ -0,0 +1,282 @@
+					resource.TestCheckResourceAttr(resourceName2, "content", "requests #test2"),
+					resource.TestCheckResourceAttr(resourceName3, "name", "testfunction1"),
+					resource.TestCheckResourceAttr(resourceName3, "scope", "shared"),
+					resource.TestCheckResourceAttr(resourceName3, "type", "function"),
+					resource.TestCheckResourceAttr(resourceName3, "content", "requests #test3"),
+					resource.TestCheckResourceAttr(resourceName3, "function_alias", "myfunction"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMApplicationInsightAnalyticsItemDestroy(queryName string, itemScope insights.ItemScope, itemType insights.ItemTypeParameter) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "azurerm_application_insights_analytics_item" {
+				continue
+			}
+			name := rs.Primary.Attributes["name"]
+			resGroup := rs.Primary.Attributes["resource_group_name"]
+
+			exists, err := testCheckAzureRMApplicationInsightsAnalyticsItemExistsInternal(rs, queryName, itemScope, itemType)
+			if err != nil {
+				return fmt.Errorf("Error checking if item has been destroyed: %s", err)
+			}
+			if exists {
+				return fmt.Errorf("Bad: Application Insights AnalyticsItem '%q' (resource group: '%q') still exists", name, resGroup)
+			}
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMApplicationInsightsAnalyticsItemExists(resourceName string, queryName string, itemScope insights.ItemScope, itemType insights.ItemTypeParameter) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+		name := rs.Primary.Attributes["name"]
+		resGroup := rs.Primary.Attributes["resource_group_name"]
+
+		exists, err := testCheckAzureRMApplicationInsightsAnalyticsItemExistsInternal(rs, queryName, itemScope, itemType)
+		if err != nil {
+			return fmt.Errorf("Error checking if item exists: %s", err)
+		}
+		if !exists {
+			return fmt.Errorf("Bad: Application Insights AnalyticsItem '%q' (resource group: '%q') does not exist", name, resGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMApplicationInsightsAnalyticsItemExistsInternal(rs *terraform.ResourceState, queryName string, itemScope insights.ItemScope, itemType insights.ItemTypeParameter) (bool, error) {
+	resGroup := rs.Primary.Attributes["resource_group_name"]
+
+	appInsightsID := rs.Primary.Attributes["application_insights_id"]
+	id, err := azure.ParseAzureResourceID(appInsightsID)
+	if err != nil {
+		return false, fmt.Errorf("Error parsing resource ID: %s", err)
+	}
+
+	appInsightsName := id.Path["components"]
+
+	conn := testAccProvider.Meta().(*ArmClient).appInsights.AnalyticsItemsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	includeContent := false
+	resp, err := conn.List(ctx, resGroup, appInsightsName, insights.AnalyticsItems, itemScope, itemType, &includeContent)
```

Comment:
can we use `conn.Get(ctx, resourceGroupName, appInsightsName, scopePath, itemID, name)` here instead?

(we should also be able to parse anything out of the state via `foo := rs.Primary.Attributes["foo"]` rather than passing them in)